### PR TITLE
AGENT-510: Support interactive network console when pxe booting 

### PR DESCRIPTION
**Pull request by pawanpinjarkar**
Code restructure. The agentImage and agentpxefiles now depend on common AgentArtifacts assets.

### CODE REVIEW

File: pkg/asset/agent/image/agentpxefiles.go [hunk 1]
```diff
@@ -34,48 +33,34 @@ var _ asset.WritableAsset = (*AgentPXEFiles)(nil)
...
 func (a *AgentPXEFiles) Generate(dependencies asset.Parents) error {
-	ignition := &Ignition{}
-	dependencies.Get(ignition)
 
-	baseImage := &BaseIso{}
+	baseImage := &BaseImage{}
 	dependencies.Get(baseImage)
 
-	a.isoPath = baseImage.File.Filename
+	a.isoPath = baseImage.baseISOFileName
```
Comment (andfasano) [inline pkg/asset/agent/image/agentpxefiles.go:46]: This, and in particular `extractPXEFileFromISO`, doesn't seem necessary anymore: the dependent asset `BaseImage` already extracted all the base iso content in a temporary folder (see `BaseImage.tmpPath`), so its content could just be simply fetched from there

File: pkg/asset/agent/image/agentimage.go [hunk 2]
```diff
@@ -32,185 +28,20 @@ var _ asset.WritableAsset = (*AgentImage)(nil)
...
-	defer initrdImg.Close()
-
-	_, err = initrdImg.Write(buff)
-	if err != nil {
-		return err
-	}
+	a.cpuArch = baseImage.cpuArch
+	a.rendezvousIP = baseImage.rendezvousIP
+	a.tmpPath = baseImage.tmpPath
+	a.volumeID = baseImage.volumeID
```
Comment (andfasano) [inline pkg/asset/agent/image/agentimage.go:44]: See previous comment for `volumeID`

File: pkg/asset/agent/image/baseimage.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+		return err
+	}
+	a.tmpPath = tmpPath
+
+	err = isoeditor.Extract(iso, a.tmpPath)
+	if err != nil {
+		return err
+	}
+
+	err = a.updateIgnitionImg(ignition)
```
Comment (andfasano) [inline pkg/asset/agent/image/baseimage.go:113]: Also this part doesn't belong here, but it is specific of `AgentImage`

File: pkg/asset/agent/image/agentimage.go
Comment (andfasano) [inline pkg/asset/agent/image/agentimage.go:1]: Given the current refactoring, I'd see a bad thing to rename this asset as `AgentISO`

File: pkg/asset/agent/image/baseimage.go [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+	rendezvousIP string
+
+	tmpPath         string
+	volumeID        string
+	baseISOFileName string
+	initrdImgPath   string
+	ignitionByte    []byte
+}
+
+var _ asset.WritableAsset = (*AgentImage)(nil)
```
[the remark this answers is not in the document]
Comment (pawanpinjarkar) [reply pkg/asset/agent/image/baseimage.go:28]: No, its an overlook

File: pkg/asset/agent/image/agentpxefiles.go (hunk 1 above)
Comment (pawanpinjarkar) [reply pkg/asset/agent/image/agentpxefiles.go:46]: Right, I see the rootfs and kernel also gets extracted to a tmp dir

File: pkg/asset/agent/image/agentimage.go [hunk 5]
```diff
@@ -22,108 +18,40 @@ const (
...
+	dependencies.Get(agentArtifacts)
 
-	err = a.updateIgnitionImg(ignition)
-	if err != nil {
-		return err
-	}
+	a.cpuArch = agentArtifacts.cpuArch
+	a.rendezvousIP = agentArtifacts.rendezvousIP
+	a.tmpPath = agentArtifacts.tmpPath
+	a.isoPath = agentArtifacts.isoPath
```
[the remark this answers is not in the document]
Comment (pawanpinjarkar) [reply pkg/asset/agent/image/agentimage.go:43]: Then the AgentImage will need to depend on BaseIso. My idea is that agentImage and agentpxefiles should just only depend on AgentArtifacts and AgentArtifacts is the place for all the dependencies.

File: pkg/asset/agent/image/agentartifacts.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+	dependencies.Get(ignition, baseIso, agentManifests, registriesConf)
+
+	ignitionByte, err := json.Marshal(ignition.Config)
+	if err != nil {
+		return err
+	}
+	a.cpuArch = ignition.CPUArch
+	a.rendezvousIP = ignition.RendezvousIP
+	a.ignitionByte = ignitionByte
+	a.isoPath = baseIso.File.Filename
```
[the remark this answers is not in the document]
Comment (pawanpinjarkar) [reply pkg/asset/agent/image/agentartifacts.go:52]: My idea is that agentImage and agentpxefiles should just only depend on AgentArtifacts and AgentArtifacts is the place for all the dependencies.

File: pkg/asset/agent/image/agentimage.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (andfasano) [reply pkg/asset/agent/image/agentimage.go]: Ok, sounds good!

**Review by rwsu — COMMENTED** [review]
/lgtm

**Pull request merged**